plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -278,6 +278,79 @@ export const createSessionRecordingEvent = (
 
     return data
 }
+
+export interface SummarizedSessionRecordingEvent {
+    uuid: string
+    timestamp: string
+    team_id: number
+    distinct_id: string
+    session_id: string
+    window_id: string | undefined
+    created_at: string
+    url: string | undefined
+    click_count: number
+    keypress_count: number
+    mouse_activity_count: number
+}
+
+interface RRWebEventSummaryData {
+    href?: string
+    source?: number
+}
+
+interface RRWebEventSummary {
+    timestamp: number
+    type: number
+    data: RRWebEventSummaryData
+}
+
+export const createSessionReplayEvent = (
+    uuid: string,
+    team_id: number,
+    distinct_id: string,
+    timestamp: DateTime,
+    ip: string | null,
+    properties: Properties
+) => {
+    const timestampString = castTimestampOrNow(timestamp, TimestampFormat.ClickHouse)
+
+    const eventsSummaries: RRWebEventSummary[] = properties['$snapshot_data']?.['events_summary'] || []
+    let clickCount = 0
+    let keypressCount = 0
+    let mouseActivity = 0
+    let url: string | undefined = undefined
+    eventsSummaries.forEach((eventSummary: RRWebEventSummary) => {
+        if (eventSummary.type === 3) {
+            mouseActivity += 1
+            if (eventSummary.data.source === 2) {
+                clickCount += 1
+            }
+            if (eventSummary.data.source === 5) {
+                keypressCount += 1
+            }
+        }
+        if (!!eventSummary.data.href?.trim().length && url === undefined) {
+            url = eventSummary.data.href
+        }
+    })
+
+    const data: SummarizedSessionRecordingEvent = {
+        uuid,
+        team_id: team_id,
+        distinct_id: distinct_id,
+        session_id: properties['$session_id'],
+        window_id: properties['$window_id'],
+        timestamp: timestampString,
+        created_at: timestampString,
```

Comment:
Do we need / care about both of these? If they are always the same....

Reply:
Instead of this I think we need to have `first_timestamp` and `last_timestamp` so that we can correctly merge all the events into one (we need to know the earliest event in the range and the latest)

Reply:
Ah, yeah, I was going to ask if you knew the difference between timestamp and created_at 🤣 

Just need one at this level and then will materialise to a view with min and max

Reply:
ugh, and `created_at` wasn't in the column listing for the materialized view so we weren't reading it into CH anyway... these "smash the strings together to make the tables" `sql.py` files are really tricksy

Reply:
> Just need one at this level and then will materialise to a view with min and max

Well, no, that's my assumption. I'll go through the app_metrics stuff to check